lib/jekyll/commands/serve.rb

```diff
@@ -1,142 +1,192 @@
-# -*- encoding: utf-8 -*-
 module Jekyll
   module Commands
     class Serve < Command
-
       class << self
+        COMMAND_OPTIONS = {
+          "detact" => ["-B", "--detach", "Run the server in the background (detach)"],
+          "port" => ["-P", "--port [PORT]", "Port to listen on"],
+          "host" => ["host", "-H", "--host [HOST]", "Host to bind to"],
+          "skip_initial_build" => ["skip_initial_build", "--skip-initial-build", "Skips the initial site build" \
+            "which occurs before the server is started."],
+
+          "enable_ssl" => ["--enable-ssl", "Enable WEBrick SSL."],
+          "ssl_cert" => ["--ssl-cert [CERT]", "X.509 (SSL) certificate."],
+          "ssl_key" => ["--ssl-key [KEY]", "X.509 (SSL) Private Key."],
+          "baseurl" => ["-b", "--baseurl [URL]", "Base URL"]
+        }
 
         def init_with_program(prog)
           prog.command(:serve) do |c|
-            c.syntax 'serve [options]'
-            c.description 'Serve your site locally'
+            c.description "Serve your site locally"
+            c.syntax "serve [options]"
             c.alias :server
             c.alias :s
 
             add_build_options(c)
+            COMMAND_OPTIONS.each do |k, v|
+              c.option k, *v
+            end
+
+            c.action do |a, o|
+              unless o.key?("watch")
+                o["watch"] = \
+                  true
+              end
 
-            c.option 'detach', '-B', '--detach', 'Run the server in the background (detach)'
-            c.option 'port', '-P', '--port [PORT]', 'Port to listen on'
-            c.option 'host', '-H', '--host [HOST]', 'Host to bind to'
-            c.option 'baseurl', '-b', '--baseurl [URL]', 'Base URL'
-            c.option 'skip_initial_build', '--skip-initial-build', 'Skips the initial site build which occurs before the server is started.'
-
-            c.action do |args, options|
-              options["serving"] = true
-              options["watch"] = true unless options.key?("watch")
-              Jekyll::Commands::Build.process(options)
-              Jekyll::Commands::Serve.process(options)
+              o["serving"] = true
+              Jekyll::Commands::Build.process(o)
+              Jekyll::Commands::Serve.process(o)
             end
           end
         end
 
         # Boot up a WEBrick server which points to the compiled site's root.
+
         def process(options)
           options = configuration_from_options(options)
-          destination = options['destination']
+          destination = options["destination"]
           setup(destination)
 
-          s = WEBrick::HTTPServer.new(webrick_options(options))
-          s.unmount("")
-
-          s.mount(
-            options['baseurl'],
-            custom_file_handler,
-            destination,
-            file_handler_options
+          server = WEBrick::HTTPServer.new(
+            webrick_options(
+              options
+            )
```

Comment:
It was a little unreadable in that it took me a few seconds to get a grip on the file which is why some of the changes landed but I also split off a bit of it to fix that problem so maybe we should review the new changes I made because they are quite a bit different.